Design/Workbench/design-arch.md

```diff
@@ -0,0 +1,241 @@
+that mimics the `pyplot` api but uses our defaults (configurable) for toolbars/window etc. Matplotlib uses a [tool][mpl-boilerplate] to generate their `pyplot` api. We can do something similar.
+
+By providing an alternate `cli` module we can preserve access to the standard `matplotlib.pyplot` module should anyone want access to this. A snippet of code such as:
+
+```python
+try:
+    import mantiqt.plotting.cli as plt
+except ImportError:
+    import matplotlib.pyplot as plt
+```
+
+would allow a script to function either using our cli or the built-in matplotlib one.
+
+### 1D Plots
+
+![1D Plot](1d-plot-window.png)
+
+### 2D Plots
+
+![2D Plot](2d-plot-window.png)
+
+These figures are meant to serve as a guide and will need to be tweaked as they are developed.
+
+### Figure Management
+
+It is desirable within the plotting framework to have a mechanism for controlling how figures are managed. For example, when `plot` is called without specifying a window then we must make a choice of either:
+
+ - create a new figure or
+ - merge/overplot with an existing figure.
+
+The initial propsosal is to borrow ideas from the [mslice][mslice] and have a controls on the figures so that users can indicate what is to be done with this window. In the figures above each figure contains a pair of
+buttons `Hold/Active` or `Held/Active` (depending on its state). These buttons control what happens when a plotting function is called without explicit intent as to which figure it should affect.
+
+- Hold: This window should be *frozen* and left as it is. Further plotting should be done in a new window. The text of the button changes to *Held* and the background colour to *red*.
+- Active: This window is the *active* window for this plot type (separate lists maintained for 1D, 2D, 3D). Further plotting will **replace** the contents of this window. The  background colour of Active goes to  *green* and the `Held/Hold` button text is set back to *Hold* and its background back to gray.
+
+The default state for new windows, i.e. whether held or active will be user configurable.
+
+### Templates
+
+Matplotlib provides a [stylesheet][mpl-stylesheets] mechanism that allows predefined styles to be applied to plots. This mechanism will be used to provide users with the ability to save a template of their plot configuration
+for future use.
+
+### Utils
+
+This will be a subpackage for the inevitable utility-type code that will be required. One such example is the `mantidqt.utils.qt.py` that will define the shim from which allow PyQt imports are performed.
+
+# Mantid Workbench
+
+A high-level structure of the workbench package is:
+
+```
+mantidworkbench
+  |-- app
+  |   |-- __init__.py
+  |   |-- start.py
+  |   |-- mainwindow.py
+  |   |-- mainmenu.py
+  |   |-- splash.py
+  |-- config
+  |   |-- __init__.py
+  |   |-- defaults.py
+  |   |-- configdialog.py
+  |   |-- plotting.py
+  |   |-- firsttimesetup.py
+  |-- resources.qrc
```

Comment:
@NickDraper I'm not sure I understand. I imagine this will also need its own `utils` package but the other concepts are not clear. Do you mean just add something to say that this is not final and can be extended as needed?